bounce/server/api/membership.py

```diff
@@ -18,42 +18,72 @@ class MembershipEndpoint(Endpoint):
 
     __uri__ = "/memberships/<club_name:string>"
 
+    @verify_token()
     @validate(GetMembershipRequest, GetMembershipResponse)
-    async def get(self, request, club_name):
+    async def get(self, request, club_name, id_from_token=None):
         """
-        Handles a GET /memberships/<club_name>?user_id=<user_id> request
+        Handles a GET /memberships/<club_name> request
         by returning the membership that associates the given user with the
         given club. If no user ID is given, returns all memberships for the
         given club.
         """
         # Decode the club name
         club_name = unquote(club_name)
-        user_id = request.args.get('user_id', None)
+        # Get the body from the JSON requests
+        body = util.strip_whitespace(request.json)
 
         # Make sure the club exists
         club_row = club.select(self.server.db_session, club_name)
         if not club_row:
             raise APIError('No such club', status=404)
-
-        # Fetch the club's memberships
-        membership_info = membership.select(
-            self.server.db_session, club_name, user_id=user_id)
+        try:
+            user_id=body.get('user_id', None),
```

Comment:
For this one I changed it to get user_id from params rather than body. But I don't know what you mean by declaring user_id before using it, I believe I am declaring it before usage?

Reply:
I moved it out of the try block, so the user_id to 'get' is retrieved from the body before the try block. What do you think about that?